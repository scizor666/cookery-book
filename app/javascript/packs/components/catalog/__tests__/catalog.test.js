import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import {recipes} from './test_data';
import RecipeCard from '../recipe_card';
import Catalog from '../catalog';

jest.mock('react-infinite-scroller', () => 'InfiniteScroll');


test('display when loading', () => {
    const wrapper = renderer.create(<Catalog/>);
    expect(wrapper.toJSON()).toMatchSnapshot();
});

test('display a catalog with products', () => {
    const wrapper = mount(<Catalog/>);
    wrapper.setState({recipes});
    expect(wrapper.find(RecipeCard).length).toBe(2);
});