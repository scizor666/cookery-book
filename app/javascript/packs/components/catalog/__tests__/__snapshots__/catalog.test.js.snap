// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display when loading 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-md-3 col-md-offset-3 container"
    >
      <div
        className="text-center"
      >
        <h1>
          My Recipes
        </h1>
        <a
          href="about:blank/recipe/new"
        >
          Add a new Recipe
        </a>
      </div>
    </div>
  </div>
  <InfiniteScroll
    hasMore={true}
    loadMore={[Function]}
    loader={
      <div
        className="loader"
      >
        Loading ...
      </div>
    }
    pageStart={0}
  >
    <div
      className="row"
    />
  </InfiniteScroll>
</div>
`;
